Update Wave docs to use `204 No Content` where appropriate (#76)
AdaGold/task-list-api: ada-project-docs/wave_03.md

then the task is updated, so that its `completed_at` value is the current date, and I get this response:

`204 No Content`

The response should have a mimetype of "application/json" to keep our API response type consistent.

> If after the `PATCH` request I fetch the task by sending a `GET` request to `/tasks/1`, I get this response:

Reviewer: Wording is just a little confusing here. Maybe something like:

```suggestion
After I have made the 'PATCH' request, I can submit a 'GET' request to 'tasks/1, which will return the response:
```

Author: This reads much easier, thanks for the suggestion!

> If after the `PATCH` request I fetch the task by sending a `GET` request to `/tasks/1`, I get this response:

Author: We keep the preceding slash and use code formatting in the other route examples through the waves, so I want to leave that in on `/tasks/1`, and it looks like we lost the ending code tic around the route, so just making a gentle modification to your suggestion.

```suggestion
After I have made the 'PATCH' request, I can submit a 'GET' request to `/tasks/1`, which will return the response:
```

then the task is updated, so that its `completed_at` value is `null`/`None`, and I get this response:

`204 No Content`

The response should have a mimetype of "application/json" to keep our API response type consistent.

> If after the `PATCH` request I fetch the task by sending a `GET` request to `/tasks/1`, I get this response:

Reviewer: Same as before, 
```suggestion
After I have made the 'PATCH' request, I can submit a 'GET' request to 'tasks/1, which will return the response:
```

> If after the `PATCH` request I fetch the task by sending a `GET` request to `/tasks/1`, I get this response:

Author: ```suggestion
After I have made the 'PATCH' request, I can submit a 'GET' request to `/tasks/1`, which will return the response:
```

then I want this to behave exactly like `/tasks/1/mark_complete` for an incomplete task. The task is updated, so that its `completed_at` value is the current date, and I get this response:

`204 No Content`

The response should have a mimetype of "application/json" to keep our API response type consistent.

> If after the `PATCH` request I fetch the task by sending a `GET` request to `/tasks/1`, I get this response:

Reviewer: Same as before. It's just a really clunky sentence to write! 
```suggestion
After I have made the 'PATCH' request, I can submit a 'GET' request to 'tasks/1, which will return the response:
```

> If after the `PATCH` request I fetch the task by sending a `GET` request to `/tasks/1`, I get this response:

Author: ```suggestion
After I have made the 'PATCH' request, I can submit a 'GET' request to `/tasks/1`, which will return the response:
```

then I want this to behave exactly like `/tasks/1/mark_incomplete` for a complete task. Its `completed_at` value remains as `null`/`None`, and I get this response:

`204 No Content`

The response should have a mimetype of "application/json" to keep our API response type consistent.

> If after the `PATCH` request I fetch the task by sending a `GET` request to `/tasks/1`, I get this response:

Reviewer: No need to take the suggestion, I'm just adding it again so its easier for you to commit the suggestions if you do!

```suggestion
After I have made the 'PATCH' request, I can submit a 'GET' request to 'tasks/1, which will return the response:```

> If after the `PATCH` request I fetch the task by sending a `GET` request to `/tasks/1`, I get this response:

Author: ```suggestion
After I have made the 'PATCH' request, I can submit a 'GET' request to `/tasks/1`, which will return the response:
```
